base/etc/osg/image-config.d/20-osg-ce-setup.sh

```diff
@@ -42,21 +42,29 @@ fi
 hostcert_path=/etc/grid-security/hostcert.pem
 hostkey_path=/etc/grid-security/hostkey.pem
 hostcsr_path=/etc/grid-security/host.req
+orig_hostcert_path=/etc/grid-security-orig.d/tls.crt
+orig_hostkey_path=/etc/grid-security-orig.d/tls.key
 
 certbot_opts="--noninteractive --agree-tos --standalone --email $CE_CONTACT -d $CE_HOSTNAME"
 
 [[ $LE_STAGING == "true" ]] && certbot_opts="$certbot_opts --dry-run"
 
 if [ ! -f $hostcert_path ] || [ ! -f $hostkey_path ]; then
-    echo "Establishing Let's Encrypt certificate.."
-    if [ -f $hostkey_path ]; then
-        openssl req -new -nodes -out $hostcsr_path -key $hostkey_path -subj "/CN=$CE_HOSTNAME"
-        certbot_opts="$certbot_opts --csr $hostcsr_path --cert-path $hostcert_path"
+    if [[ -f $orig_hostcert_path && -f $orig_hostkey_path ]]; then
+        echo "Using host cert/key mounted in /etc/grid-security-orig.d/"
+        ln -s $orig_hostcert_path $hostcert_path
+        ln -s $orig_hostkey_path $hostkey_path
```

Comment:
These two `ln` commands are _both_ run if _either_ hostcert or hostkey is missing.

Is there a relevant use case where only one is provided?  If so, you might want to add the `[ -f $hostcert_path ] || ln -s ...` as you have below.  But if you expect either both or neither to be there then it doesn't really matter i guess.

Reply:
>  But if you expect either both or neither to be there then it doesn't really matter i guess.

Yeah, if either the target host cert/key exist AND either the "orig" host cert/key exist that's an error and since we `set -e` the symlink failure should bail us out of the setup.